lib/atmega328/render.cpp

```diff
@@ -0,0 +1,147 @@
+#include "render.h"
+
+#define DIGITS_X_OFFSET 2
+#define TOP_Y_OFFSET 0
+#define BOTTOM_Y_OFFSET 8
+#define ACTIVE_INDICATOR_BLINK_RATE 200
+#define PAUSED_TIMER_BLINK_RATE 500
+#define TIMERS_INDICATOR_COLUMN 0
+#define FONT_WIDTH 6
+#define FONT_HEIGHT 7
+
+static uint16_t last_rendered_time;
+static state_t last_rendered_state;
+static bool last_active_timer_indicator_blink_is_on;
+static bool last_paused_timer_blink_is_on;
+
+void init_render()
+{
+    matrix_init();
+    init_millis();
+}
+
+static bool is_active_timer_indicator_blink_on()
+{
+    static bool is_on = true;
+    uint16_t millis_now = millis();
+    static uint16_t last_active_timer_indicator_blink_time;
+
+    if ((millis_now - last_active_timer_indicator_blink_time) >= ACTIVE_INDICATOR_BLINK_RATE) {
+        last_active_timer_indicator_blink_time = millis_now;
+        is_on = !is_on;  // toggle blinking
+    }
+
+    return is_on;
+}
+
+static void draw_timers_indicator(const state_machine_t timers[], uint8_t num_timers, uint8_t active_timer_index, bool active_timer_indicator_state)
+{
+    for (uint8_t i = 0; i < num_timers && i < MAX_TIMERS; i++)
+    {
+        bool is_idle = timers[i].state == IDLE;
+        bool is_running = timers[i].state == RUNNING;
+        bool is_paused = timers[i].state == PAUSED;
+        bool is_ringing = timers[i].state == RINGING;
+        bool is_active = (i == active_timer_index);
+
+        bool show_led = is_idle || is_running || is_paused || is_ringing;
+
+        if (is_active) {
+            show_led = show_led && active_timer_indicator_state;
+        }
+
+        matrix_set_pixel(TIMERS_INDICATOR_COLUMN, i, show_led);
+    }
+
+}
+
+static void draw_digit(char digit, uint8_t x_offset, uint8_t y_offset)
+{
+    const uint8_t* ptr_digit = get_char(digit);
+    
+    for (uint8_t row = 0; row < FONT_HEIGHT; row++)
+    {
+        for (uint8_t col = 0; col < FONT_WIDTH; col++)
+        {
+            bool is_on = ptr_digit[row] & (1 << (5 - col));  // 6-bit wide
```

Comment:
Replace the comment with `FONT_WIDTH` and the `bit` function

```suggestion
            bool is_on = ptr_digit[row] & bit(FONT_WIDTH - 1 - col);
```

Reply:
done